src/Model/Profile.php

```diff
@@ -334,7 +343,7 @@ private static function sidebar(App $a, array $profile, $block = 0, $show_connec
 
 			if (Contact::canReceivePrivateMessages($profile)) {
 				if ($visitor_is_followed || $visitor_is_following) {
-					$wallmessage_link = $visitor_base_path . '/message/new/' . base64_encode($profile['addr'] ?? '');
+					$wallmessage_link = $visitor_base_path . '/message/new/' . $profile['cid'];
```

Comment:
This part is something I want to have a closer look at.

Reply:
Does this prevent this PR from being merged?

Reply:
Yeah. I really want to have a close look at all the consequences.